src/components/utils/validation.test.js

```diff
@@ -36,3 +36,69 @@ describe('check VM names', function () {
     expect(isVmNameValid('veryveryveryveryveryveryveryveryveryveryveryveryveryveryverylongname')).toEqual(false)
   })
 })
+
+describe('check NIC names', function () {
+  it('valid NIC names', function () {
+    expect(isNicNameValid('SomeName')).toEqual(true)
+    expect(isNicNameValid('1Ab')).toEqual(true)
+    expect(isNicNameValid('A_c')).toEqual(true)
+    expect(isNicNameValid('c1-b')).toEqual(true)
+    expect(isNicNameValid('A10-_.b')).toEqual(true)
+  })
+  it('invalid NIC names ', function () {
+    expect(isNicNameValid('SomeBadName?')).toEqual(false)
+    expect(isNicNameValid('The Worst name Ever!')).toEqual(false)
+    expect(isNicNameValid('')).toEqual(false)
+    expect(isNicNameValid('N@me$)')).toEqual(false)
+    expect(isNicNameValid('A1 b1')).toEqual(false)
+  })
+})
+describe('check NIC name uniqueness', function () {
+  const nicList = [
+    {
+      deviceType: 'virtio',
+      id: '12fe1e4e-176a-4bde-8336-c73fcad13114',
+      isFromTemplate: true,
+      name: 'nic1',
+      vnicProfileId: '0000000a-000a-000a-000a-000000000398',
+    },
+    {
+      deviceType: 'rtl8139',
+      id: 'NEW_0ff77677-94ff-4088-a141-29033f838813',
+      name: 'nic2',
+      vnicProfileId: '7b7c87a4-441b-4116-a2f6-27dd0331dfdf',
+    },
+    {
+      deviceType: 'e1000',
+      id: 'NEW_6a209728-aa22-4313-8138-62ce4790793e',
+      name: 'nic3',
+      vnicProfileId: '1d32a4e5-7851-43e1-903c-25c1fd0a792d',
+    },
+    {
+      deviceType: 'virtio',
+      id: 'NEW_3849c219-bc68-4c3e-a851-a7f6e0bc4ae1',
+      name: 'nic4',
+      vnicProfileId: 'af4d8ef3-b379-4044-832b-cb086e1384e4',
+    },
+  ]
+  const createdDuplicateName = { deviceType: 'virtio', id: 'NEW_3849c219-bc68-4c3e-a851-a7f6e0tsv9l3', name: 'nic4', vnicProfileId: '1d32a4e5-7851-43e1-903c-25c1fd0a792d' }
+  const editedUniqueNewName = { deviceType: 'virtio', id: 'NEW_3849c219-bc68-4c3e-a851-a7f6e0bc4ae1', name: 'nic5', vnicProfileId: 'af4d8ef3-b379-4044-832b-cb086e1384e4' }
+  const createUniqueName = { deviceType: 'virtio', id: 'NEW_3849c219-bc68-4c3e-a851-a7f6e0tsv9l3', name: 'nic5', vnicProfileId: '1d32a4e5-7851-43e1-903c-25c1fd0a792d' }
+  const editedDuplicate = { deviceType: 'virtio', id: 'NEW_3849c219-bc68-4c3e-a851-a7f6e0bc4ae1', name: 'nic3', vnicProfileId: 'af4d8ef3-b379-4044-832b-cb086e1384e4' }
+  const editedUniqueSameName = nicList[3]
+  it('duplicate name on create new NIC', function () {
+    expect(isNicNameUnique(nicList, createdDuplicateName)).toEqual(false)
```

Comment:
Tests look ok, but I wonder if a different format would make them easier to read.  Something like:
```js
    expect(isNicNameUnique(nicList, { id: 'A', name: 'nic4' })).toEqual(false)
```

assuming adjustments to `nicList` more like this, since it's easier to look at test data with simple ids:
```js
const nicList = [
  { id: '1', name: 'nic1' },
  { id: '2', name: 'nic2' },
  { id: '3', name: 'nic3' },
  { id: '4', name: 'nic4' },
]
```

The other tests can be adjusted in the same way.

Reply:
I agree, my first thought was to use real data, but it really makes it clearer